packages/server-wallet/src/protocols/__test__/ledger-manager.test.ts

```diff
@@ -382,54 +430,162 @@ describe('exchanging ledger proposals', () => {
   });
 
   describe('as responding proposer', () => {
-    it('will propose identical proposal to counterparty with same requests', () => {
-      const requestChannel = prefundChannelWithAllocations([alice, 1]);
-      const protocolArgs = {
-        fundingChannel: defaultLedgerChannel,
-        channelsRequestingFunds: [requestChannel.protocolState],
-        theirLedgerProposal: {
-          proposal: simpleEthAllocation([alice, 9], [bob, 10], [requestChannel.channelId, 1]),
-          nonce: 0,
-        },
-      };
-      expect(protocol(processLedgerQueueProtocolState(protocolArgs))).toMatchObject({
+    it('will propose identical proposal to counterparty with same requests', async () => {
+      // create an application channel
+      const appChannel = TestChannel.create({aBal: 1, bBal: 0});
+      await appChannel.insertInto(store, {states: [0, 1]});
+
+      // create and insert a funded ledger channel that doesn't fund any channels yet. Note distinct channel nonce ensured automatically
+      const ledgerChannel = TestLedgerChannel.create({});
+      await ledgerChannel.insertInto(store, {
+        states: [3, 4],
+        bals: [10, 10],
+      });
+
+      const bobsProposal = simpleEthAllocation([alice, 9], [bob, 10], [appChannel.channelId, 1]);
+      // add a proposal from bob into Alice's store
+      await ledgerChannel.insertProposal(
+        store,
+        bobsProposal,
+        ledgerChannel.participantB.signingAddress,
+        0
+      );
+
+      // add consistent funding requests to Alice's store
+      await ledgerChannel.insertFundingRequest(store, appChannel.channelId);
+
+      // crank the ledger manager
+      const response = new WalletResponse();
+      await ledgerManager.crank(ledgerChannel.channelId, response);
+      await ledgerManager.crank(ledgerChannel.channelId, response);
+      await ledgerManager.crank(ledgerChannel.channelId, response);
+      await ledgerManager.crank(ledgerChannel.channelId, response);
```

Comment:
Presumably this should become just one crank when the refactor ships?

Reply:
Actually I think these repeated cranks are a debugging artefact. I've removed them and the test still passes.